File: DynamicKey/AgoraDynamicKey/deno/sample/SttTokenBuilderSample.js
import { SttTokenBuilder } from '../src/SttTokenBuilder.js'
import { Role } from '../src/RtcTokenBuilder2.js'

const appId = Deno.env.get('AGORA_APP_ID')
const appCertificate = Deno.env.get('AGORA_APP_CERTIFICATE')
const channelName = '7d72365eb983485397e3e3f9d460bdda'
const rtcAccount = '2882341273'
const rtcRole = Role.PUBLISHER
const rtcTokenExpire = 3600
const joinChannelPrivilegeExpire = 3600
const pubAudioPrivilegeExpire = 3600
const pubVideoPrivilegeExpire = 3600
const pubDataStreamPrivilegeExpire = 3600
const rtmUserId = '2882341273'
const rtmTokenExpire = 3600

console.log('App Id:', appId)
if (!appId || !appCertificate) {
  console.log('Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE')
  Deno.exit(0)
}

const token = SttTokenBuilder.buildToken(
  appId,
  appCertificate,
  channelName,
  rtcAccount,
  rtcRole,
  rtcTokenExpire,
  joinChannelPrivilegeExpire,
  pubAudioPrivilegeExpire,
  pubVideoPrivilegeExpire,
  pubDataStreamPrivilegeExpire,
  rtmUserId,
  rtmTokenExpire,
)
console.log('STT token:', token)

File: DynamicKey/AgoraDynamicKey/cpp/sample/SttTokenBuilderSample.cpp
// Copyright (c) 2014-2026 Agora.io, Inc.

#include <cstdlib>
#include <iostream>

#include "../src/SttTokenBuilder.h"

using namespace agora::tools;

int main() {
  const char *env_app_id = getenv("AGORA_APP_ID");
  std::string app_id = env_app_id ? env_app_id : "";
  const char *env_app_certificate = getenv("AGORA_APP_CERTIFICATE");
  std::string app_certificate = env_app_certificate ? env_app_certificate : "";

  std::cout << "App Id:" << app_id << std::endl;
  if (app_id.empty() || app_certificate.empty()) {
    std::cout << "Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE" << std::endl;
    return -1;
  }

  std::string token = SttTokenBuilder::BuildToken(
      app_id, app_certificate, "stt-channel", "stt-rtc-user", UserRole::kRolePublisher,
      3600, 3600, 3600, 3600, 3600, "stt-rtm-user", 3600);
  std::cout << "STT token:" << token << std::endl;
  return 0;
}
